parser/src/main/scala/jawn/Parser.scala

```diff
@@ -90,6 +90,23 @@ abstract class Parser[J] {
   protected[this] def die(i: Int, msg: String): Nothing =
     die(i, msg, ErrorContext)
 
+  /**
+   * A version of `at` that returns as much of the indicated substring as
+   * possible without throwing exceptions.
+   *
+   * This method is necessary because `at(i, j)` may fail with an exception even
+   * when `atEof(i, j)` returns false (see #143 for an example). This method is
+   * a workaround; it is not optimized or robust against invalid input and
+   * should not be used outside of the context of `die`.
+   */
+  private[this] def safeAt(i: Int, j: Int): CharSequence =
+    if (j <= i) "" else {
+      try at(i, j) catch {
+        case _: Exception =>
+          safeAt(i, j - 1)
```

Comment:
`j-i` is at most `ErrorContext` (which is 6), right?  Just wondering about catching and throwing absurd numbers of exceptions as it walks back, but I won't lose sleep over up to 6.  And I think it's `@tailrec`, though I didn't ask scalac.

Reply:
In theory you could pass some ridiculously large context to `die`, but it's protected, so I'm not too worried about that. Also except in the (kind of weird) `AsyncParser` usage the `at` call in `safeAt` will never throw exceptions at all, since `atEof` has been checked.

(It is `@tailrec`, by the way, so I went ahead and added it.)